internal/core/processor/evolution/processor_test.go

```diff
@@ -0,0 +1,482 @@
+
+	"github.com/freeverseio/laos-universal-node/internal/scan"
+	mockTx "github.com/freeverseio/laos-universal-node/internal/state/mock"
+	"go.uber.org/mock/gomock"
+)
+
+func TestGetInitStartingBlock(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                           string
+		startingBlockData              model.Block
+		startingBlockError             error
+		userProvidedBlock              uint64
+		LastBlockNumberFromClient      uint64
+		LastBlockNumberFromClientError error
+		expectedResult                 uint64
+		expectedError                  error
+	}{
+		{
+			name:               "starting block exists in storage",
+			startingBlockData:  model.Block{Number: 10},
+			startingBlockError: nil,
+			userProvidedBlock:  0,
+			expectedResult:     11,
+			expectedError:      nil,
+		},
+
+		{
+			name:               "starting block does not exist in storage, it returns error",
+			startingBlockData:  model.Block{},
+			startingBlockError: errors.New("error from storage"),
+			userProvidedBlock:  0,
+			expectedResult:     0,
+			expectedError:      errors.New("error retrieving the current block from storage: error from storage"),
+		},
+
+		{
+			name:              "starting block does not exist in storage, user provided starting block",
+			startingBlockData: model.Block{},
+			userProvidedBlock: 20,
+			expectedResult:    20,
+			expectedError:     nil,
+		},
+		{
+			name:                           "starting block does not exist in storage, user provided starting block is zero",
+			startingBlockData:              model.Block{},
+			startingBlockError:             nil,
+			userProvidedBlock:              0,
+			LastBlockNumberFromClient:      30,
+			LastBlockNumberFromClientError: nil,
+			expectedResult:                 30,
+			expectedError:                  nil,
+		},
+
+		{
+			name:                           "starting block does not exist in storage, user provided starting block is zero, error from client",
+			startingBlockData:              model.Block{},
+			startingBlockError:             nil,
+			userProvidedBlock:              0,
+			LastBlockNumberFromClient:      0,
+			LastBlockNumberFromClientError: errors.New("error from client"),
+			expectedResult:                 0,
+			expectedError:                  errors.New("error retrieving the latest block from chain: error from client"),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.TODO()
+			stateService, tx, client, _ := createMocks(t)
+
+			stateService.EXPECT().NewTransaction().Return(tx)
+			tx.EXPECT().GetLastEvoBlock().Return(tt.startingBlockData, tt.startingBlockError)
+			tx.EXPECT().Discard()
+			if tt.userProvidedBlock == 0 && tt.startingBlockData.Number == 0 && tt.startingBlockError == nil {
+				client.EXPECT().BlockNumber(ctx).Return(tt.LastBlockNumberFromClient, tt.LastBlockNumberFromClientError)
+			}
```

Comment:
If you like, you can get rid of this `if` by adding another parameter in the test struct called `blockNumberTimes` and then appending `.Times(tt.blockNumberTimes)` to `client.EXPECT()...`.
The same thing applies to `TestVerifyChainConsistency`

Reply:
If you are not insisting I will keep it as is. I usually accept up to 2 `ifs`` in table test. 

Also I don't like to many parametters in the test. every parameter I add I have a feeling that I have to double the number of tests because the number of combinations doubles. 
This is more my personal feelings but not so far from reality.

Reply:
I won't insist, your reasoning has convinced me